.github/workflows/job-build.yml

```
@@ -104,7 +104,12 @@ jobs:
         name: "Run: make binaries with custom BUILDTAGS"
         run: |
           set -eux
-          # no_ipfs: make sure it does not incur any IPFS-related dependency
+          # Disabling all optional features and ensuring this does not incur any of the related dependencies
           go mod vendor
-          rm -rf vendor/github.com/ipfs vendor/github.com/multiformats
-          BUILDTAGS=no_ipfs make binaries
+          rm -rf \
+            vendor/github.com/ipfs \
+            vendor/github.com/multiformats \
+            vendor/github.com/containerd/nydus-snapshotter \
+            vendor/github.com/containerd/stargz-snapshotter \
+            vendor/github.com/containerd/accelerated-container-image
+          BUILDTAGS=no_ipfs,no_nydus,no_esgz,no_obd make binaries
```

Tags should be named `no_${SNAPSHOTTER_NAME}`
```suggestion
          BUILDTAGS=no_ipfs,no_nydus,no_stargz,no_overlaybd make binaries
```

---

esgz -> stargz - I was wondering - we might want to do that generally (eg: on var and func names as well).

---

Maybe yes, but it can be discussed separately, as var names and func names are not visible to users and package maintainers.